zio-uring/src/main/scala/RingIO.scala

```diff
@@ -27,7 +61,20 @@ class RingIO(uring: Ring) {
 }
 
 object RingIO {
-  def make(queueSize: Int, completionsChunkSize: Int): IO[IOException, RingIO] = IO.succeed {
-    new RingIO(Ring.make(queueSize, completionsChunkSize))
+  def make(queueSize: Int, completionsChunkSize: Int, shutdown: AtomicBoolean): IO[IOException, RingIO] = IO.succeed {
+    new RingIO(Ring.make(queueSize, completionsChunkSize, shutdown))
+  }
+
+  def managed(queueSize: Int, completionsChunkSize: Int): ZManaged[Any, IOException, RingIO] = {
+    val shutdown = new AtomicBoolean(false)
+    val effect   = for {
+      ring   <- make(queueSize, completionsChunkSize, shutdown)
+      poller <- ring.poll().repeatUntil(_ => shutdown.get()).fork
```

Comment:
You could do:
```scala
ring <- make(...).toManaged(_.close().orDie)
poller <- ring.poll().forever.forkManaged
```

This will interrupt the fiber as part of the ZManaged release.

Reply:
Yeah, the `AtomicBoolean` mainly exists to fence those operations. But I think we don't need to pass it in externally to do that.